mlflow/transformers/__init__.py

```diff
@@ -456,6 +468,17 @@ def save_model(
         _save_example(mlflow_model, input_example, str(path), example_no_conversion)
     if metadata is not None:
         mlflow_model.metadata = metadata
+    if prompt_template:
+        # prevent saving prompt templates for unsupported pipeline types
+        if built_pipeline.task not in _SUPPORTED_PROMPT_TEMPLATING_TASK_TYPES:
+            supported_tasks_repr = ", ".join(_SUPPORTED_PROMPT_TEMPLATING_TASK_TYPES)
+            raise MlflowException(
+                f"Prompt templating is not supported for the `{built_pipeline.task}` task type. "
+                f"Supported task types are: {supported_tasks_repr}."
+            )
+
+        _validate_prompt_template(prompt_template)
+        mlflow_model.prompt_template = prompt_template
```

Comment:
Can we reuse metadata instead? prompt_template is not a param that's applicable to all Models

Reply:
Ah good idea, yeah that makes sense!